packages/fluentui/react-northstar/src/components/Tree/Tree.tsx

```diff
@@ -96,129 +104,133 @@ export interface TreeItemForRenderProps {
-            }
-          });
-        };
-
-        iterateItems(items);
+const expandedItemsGenerator = (items, acc = []) =>
+  _.reduce(
+    items,
+    (acc, item) => {
+      if (item['expanded'] && acc.indexOf(item['id']) === -1) {
+        acc.push(item['id']);
       }
-    }
 
-    if (!activeItemIds && items) {
-      activeItemIds = prevState.activeItemIds;
-
-      const expandedItemsGenerator = (items, acc = activeItemIds) =>
-        _.reduce(
-          items,
-          (acc, item) => {
-            if (item['expanded'] && acc.indexOf(item['id']) === -1) {
-              acc.push(item['id']);
-            }
-
-            if (item['items']) {
-              return expandedItemsGenerator(item['items'], acc);
-            }
+      if (item['items']) {
+        return expandedItemsGenerator(item['items'], acc);
+      }
 
-            return acc;
-          },
-          acc,
-        );
+      return acc;
+    },
+    acc,
+  );
+
+const iterateItems = (items, acc = []) =>
+  _.reduce(
+    items,
+    (acc, item) => {
+      if (item['selected'] && acc.indexOf(item['id']) === -1) {
+        acc.push(item['id']);
+      }
+      if (item['items']) {
+        return iterateItems(item['items']);
+      }
 
-      expandedItemsGenerator(items);
-    }
+      return acc;
+    },
+    acc,
+  );
+
+const Tree: React.FC<WithAsProp<TreeProps>> &
+  FluentComponentStaticProps<TreeProps> & {
+    Item: typeof TreeItem;
+    Title: typeof TreeTitle;
+  } = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(Tree.displayName, context.telemetry);
+  setStart();
+
+  const {
+    exclusive,
+    items,
+    renderItemTitle,
+    selectable,
+    children,
+    renderedItems,
+    className,
+    design,
+    styles,
+    variables,
+  } = props;
+
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(Tree.handledProps, props);
+
+  const [activeItemIds, setActiveItemIdsState] = useAutoControlled({
+    defaultValue: props.defaultActiveItemIds,
+    value: props.activeItemIds,
+    initialValue: expandedItemsGenerator(items),
+  });
+
+  const [selectedItemIds, setSelectedItemIdsState] = useAutoControlled({
+    defaultValue: props.defaultSelectedItemIds,
+    value: props.selectedItemIds,
+    initialValue: iterateItems(items),
+  });
+
+  const getA11yProps = useAccessibility<TreeBehaviorProps>(props.accessibility, {
+    debugName: Tree.displayName,
+    rtl: context.rtl,
+  });
+
+  const { classes } = useStyles<TreeStylesProps>(Tree.displayName, {
+    className: treeClassName,
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
 
-    return {
-      activeItemIds,
-      selectedItemIds,
-    };
-  }
+  const treeRef = React.useRef<HTMLElement>();
+  const itemsRef = new Map<string, React.RefObject<HTMLElement>>();
```

Comment:
I already mentioned that this thing is unstable and will be recreated between rerenders

```suggestion
  const itemsRef = React.useRef(new Map<string, React.RefObject<HTMLElement>>());
```